### ISSUE TITLE
Sorting on facet in recursive query doesn't return predicates?

### ISSUE THREAD

**Issue by lazharichir** [dgraph-io/dgraph#1642]
The problem I have is that sorting my facet (weight) on an edge results in the string values of other predicates not being returned (but the nodes are returned.)
```
# THIS WORKS
{
  recurse( func: eq(xid, "Algeria"), depth: 5 ) {
    _uid_
    xid
    related @facets(weight)
  }
}

```
And returns [good.txt](https://github.com/dgraph-io/dgraph/files/1388027/good.txt)

```
# THIS GENERATES THE PROBLEM (FILTERING ADDED)
{
  recurse( func: eq(xid, "Algeria"), depth: 5 ) {
    _uid_
    xid
    related @facets(orderdesc: weight)
  }
}
```
And returns [bad.txt](https://github.com/dgraph-io/dgraph/files/1388026/bad.txt)

Additionally, it seems like the very same query is returning a different number of nodes (total at the bottom of each file) - and the query is the same, just one has sorting applied.

I run 0.8.3 on macOS.

**Comment by peterstace** [comment]
This looks like a bug, but I haven't had any luck reproducing it so far. Is it possible for you to share a reproducible example? A minimal example that triggers the bug would be ideal, but otherwise the whole dataset would be good as well.

**Comment by lazharichir** [comment]
@peterstace I tried to make it as simple as possible by breaking it down into parts that I actually generated to insert the data.

Be careful for 2 and 3 since the UID of the starting node is hardcoded towards the end of each file so make sure you get the ID of Algeria (step 2), and then of Berber_languages (last file, the step 3.) 

[0-schema.txt](https://github.com/dgraph-io/dgraph/files/1389986/0-schema.txt)
[1-initial.txt](https://github.com/dgraph-io/dgraph/files/1389987/1-initial.txt)
[2-related to algeria.txt](https://github.com/dgraph-io/dgraph/files/1389988/2-related.to.algeria.txt)
[3-related to Berber_languages.txt](https://github.com/dgraph-io/dgraph/files/1389989/3-related.to.Berber_languages.txt)

And the query:
```
{
  recurse( func: eq(xid, "Algeria"), depth: 5 ) {
    _uid_
    xid
    related @facets(orderdesc: weight)
  }
}
```

**Comment by peterstace** [comment]
Thanks for providing the data. I've managed to reproduce the problem. Oddly though, I don't see it occur when using `orderdesc`. It only shows up for me when using `orderasc`.

**Comment by lazharichir** [comment]
@peterstace odd.. it occurs in both sorting scenarios and also when I go deep (with my full dataset) without sorting, it hides the predicates in the last "leaves"

**Comment by peterstace** [comment]
Minimal example which exhibits (what I suspect is) the root issue:

```
mutation{
	set{
		<0x1> <link> <0x2> (weight=0.4) .
		<0x1> <link> <0x3> (weight=0.5) .
		<0x2> <xid> "a" .
		<0x3> <xid> "b" .
	}
}
{
	recurse(func: uid(0x1)) {
		_uid_
		xid
		link @facets(orderdesc: weight)
	}
}
```

gives:
```
{
  "extensions": {
    "server_latency": {
      "json": "5ms",
      "parsing": "93µs",
      "processing": "220µs",
      "total": "5ms"
    }
  },
  "data": {
    "recurse": [
      {
        "_uid_": "0x1",
        "link": [
          {
            "_uid_": "0x3",
            "@facets": {
              "_": {
                "weight": 0.5
              }
            }
          },
          {
            "_uid_": "0x2",
            "@facets": {
              "_": {
                "weight": 0.4
              }
            }
          }
        ]
      }
    ]
  }
}
```

What seems to be happening is when data is sorted by facets and the resultant ordering is different than the UID ordering (in this case the ordering is opposite), then some value data (in this case xid) is lost in the result.

**Comment by peterstace** [comment]
PR that will fix this: $#1653

**Comment by pawanrawal** [comment]
A fix was merged to `master` and `release/v0.8.3`.
